spec/h2_connection_spec.lua

```diff
@@ -157,6 +159,40 @@ describe("http2 connection", function()
 			assert.truthy(cq:empty())
 		end)
 	end)
+	describe("priority on missing stream", function()
+		it("sets default priority for streams with missing parent", function()
+			local cq = cqueues.new()
+			local c, s = cs.pair()
+			cq:wrap(function()
+				c = h2_connection.new(c, "client")
+				local client_stream = c:new_stream()
+				local req_headers = new_headers()
+				req_headers:append(":method", "GET")
+				req_headers:append(":scheme", "http")
+				req_headers:append(":path", "/")
+				assert(client_stream:write_headers(req_headers, false))
```

Comment:
Gotcha. I have a couple of socket leak bugfixes for server afterwards, but it needs more work to submit it in neater patches.

I wasn't thinking about `__gc`, just having a loop in `h2_connection.close` to go over all streams in `self.streams` and calling `shutdown()` on those not in closed state.